rest-api-spec/src/main/resources/rest-api-spec/test/search/390_search_as_you_type.yml

```diff
@@ -0,0 +1,63 @@
+setup:
+  - do:
+      indices.create:
+        index:  test_1
+        body:
+          mappings:
+            properties:
+              text:
+                type: search_as_you_type
+                fields:
+                  subField:
+                    type: keyword
+  - do:
+      index:
+        index:  test_1
+        id:     1
+        body:   { text: test search as you type }
+
+  - do:
+      indices.refresh:
+        index: [test_1]
+
+---
+teardown:
+  - do:
+      indices.delete:
+        index: test_1
+
+# related issue: https://github.com/opensearch-project/OpenSearch/issues/5035
+---
+"Test search_as_you_type data type supports multi-fields":
+  - skip:
+      version: " - 2.99.99"
+      reason: "the bug was fixed since 3.0.0"
+
+  - do:
+      indices.get_mapping: {
+        index: test_1
+      }
+
+  - match: {test_1.mappings.properties.text.type:     search_as_you_type}
+  - match: {test_1.mappings.properties.text.fields.subField.type:     keyword}
+
+  - do:
+      search:
+        index: test_1
+        body:
+          query:
+            multi_match:
+              query: "test search"
+              type: "bool_prefix"
+
+  - match: {hits.total.value: 1}
+
+  - do:
+      search:
+        index: test_1
+        body:
+          query:
+            term:
```

Comment:
I am wondering if the term query is really a fit here since we are not checking multi_match, may be:

```
          multi_match:
              query: "test search"
              type: "bool_prefix"
              fields:
                - text.subField
```

Reply:
`text.subField` is keyword type, here we just verify that this field exists or not, using a simple `term` query is enough. The `multi_match` query case above is to verify the `search_as_you_type` type field `text` can work, since a `multi_match` query of type `bool_prefix` suits for the `search_as_you_type` type field.

Reply:
Correct, the issue (please correct me if I am wrong) was that multimatch on subfield (`text.subField`) didn't work, the test is perfectly valid, I was just curious it tests what was fixed? Or we need more tests?

Reply:
The issue is that the subfield doesn't exist in the mapping, so we cannot use that field `text.subField` to do anything. Adding more tests makes sense, I've added a case that tests multi_match query against the subfield.

Reply:
:+1: thank you!